tests/test_removers_listed_value_remover.py

```diff
@@ -38,9 +53,32 @@ def test_remover():
     )
 
 
-# Tests
-def test_remove_listed_value_from_end_of_feature(test_remover):
-    removed_value_information = test_remover.remove_listed_value(
+def test_remove_listed_value(test_remover):
+    gs_removed_value_information = {
+        "feature_id": "A-9",
+        "en": "No diphthongs and triphthongs",
+        "ru": "Дифтонги и трифтонги отсутствуют",
+    }
+
+    assert test_remover.remove_listed_value("A-9-1") == gs_removed_value_information
+
+    check_existence_of_output_csv_file_and_compare_with_gold_standard(
+        output_file=test_remover.output_file_with_listed_values,
+        gold_standard_file=GS_FILE_WITH_LISTED_VALUES_REMOVING_A_9_1,
+    )
+
+    (test_remover.output_dir_with_feature_profiles / "corsican.csv").unlink()
+    (test_remover.output_dir_with_feature_profiles / "ukrainian.csv").unlink()
+    (
+        test_remover.output_dir_with_feature_profiles
+        / "changed_profiles_A-9-1_No_diphthongs_and_triphthongs.txt"
+    ).unlink()
```

Comment:
is it really necessary? `check_existence...` already unlinks files.

I don't think you need to unlink the output directory. Anyway, you have to test both cases in some separate test function: when the output directory doesn't exist and when it already exists. 

Since I suggested to move creation of non-existent output dir to `__init__()` method of `ListedValueRemover`, a test can be run for that method. And here, it's OK to assume that the output dir already exists.

Not sure I made myself clear here :)

Reply:
> is it really necessary? `check_existence...` already unlinks files.

At the time when I worked on this method, for an unknown reason these files persistently remained after applying tests. Now, I have tried to comment these lines out and there were no extra files left after the tests, which is quite strange to me (but I erased these lines anyway).

> Anyway, you have to test both cases in some separate test function: when the output directory doesn't exist and when it already exists.

But, if I am not mistaken, in actual practice with the DB the input directory is always the output directory too so the output directory always exists and checking its existence is needed for tests only. Should I write the test anyway?

> Since I suggested to move creation of non-existent output dir to `__init__()` method of `ListedValueRemover`

Introducing `__init__` (with calling `super.__init__()` inside) causes an error in every test:
```
tests/test_removers_listed_value_remover.py::test__remove_from_feature_profiles - TypeError: ListedValueRemover.__init__() got an unexpected keyword argument 'input_file_with_listed_values'
```
How could it possibly be fixed?

Reply:
I have also thought once again of introducing `remove()` to this method (https://github.com/Jazyki-Mira/langworld_db_data/pull/301#discussion_r1841732054) but could not come up with any ways to do it. However I found a way to make the `enumerate()` cycle shorter

Reply:
closing this thread as these lines were removed